src/App.tsx

```diff
@@ -1,19 +1,113 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect, useMemo, useRef, useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
 
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Error } from './components/Error/Error';
+
+import { Todo } from './types/Todo';
+import { TypeError } from './types/TypeError';
+import { FilterType } from './types/FilterType';
+import { NewTodoField } from './components/NewTodoField/NewTodoField';
+import { getTodos } from './api/todos';
+
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const [hasError, setHasError] = useState(false);
+  const [errorType, setErrorType] = useState<TypeError>(TypeError.NONE);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  async function LoadTodos() {
+    if (user) {
+      try {
+        const reponse = await getTodos(user.id);
+
+        setTodos(reponse);
+      } catch {
+        setHasError(true);
+      }
+    }
+  }
+
+  useEffect(() => {
+    LoadTodos();
+  }, []);
+
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  const activeTodosLength = todos.filter(todo => todo.completed === false)
+    .length;
+
+  const onInputChange = (str: string) => {
+    setQuery(str);
+  };
+
+  const onFormSubmit = () => {
+    if (query.trim()) {
+      const newTodo: Todo = {
+        id: +new Date(),
+        userId: user?.id || +new Date(),
+        title: query.trim(),
+        completed: false,
+      };
+
+      setTodos((prevTodos) => [...prevTodos, newTodo]);
+    }
+
+    setQuery('');
+  };
+
+  const onDeleteTodo = (id: number) => {
+    setTodos(todos.filter(todo => todo.id !== id));
+  };
+
+  const onDeleteCompletedTodos = () => {
+    setTodos(todos.filter(todo => todo.completed !== true));
+  };
+
+  const removeErrorHandler = () => {
+    setHasError(false);
+    setErrorType(TypeError.NONE);
+  };
+
+  if (hasError) {
+    setTimeout(() => setHasError(false), 3000);
+  }
+
+  function getFilteredGoods() {
+    return todos.filter(todo => {
+      switch (filterType) {
+        case FilterType.Active:
+          return todo.completed === false;
+
+        case FilterType.Completed:
+          return todo.completed === true;
+
+        default:
+          return todo;
+      }
+    });
+  }
+
+  const visibleTodos = useMemo(() => {
+    return getFilteredGoods();
```

Comment:
There is not necessary to create a separate `getFilteredGoods` function if you use it just once. You can filter todos directly inside this `useMemo` hook.